### ISSUE TITLE
C++ | use protobuf as part of a project using cmake 

### ISSUE THREAD

**Issue by yonatan555** [protocolbuffers/protobuf#10971]
I am working on a project which integrate protobuf, OS: Centos 7 Linux , c++ GCC compiler v9, VM.
V3.21.9 version

work with OTLP project - Protobuf is required, while run cmake at OTLP project protobuf is failed.

this what I got:

`CMake Error at /usr/local/share/cmake-3.24/Modules/FindProtobuf.cmake:493 (file):
  file STRINGS file
  "/root/protobuf-3.21.9/src/google/protobuf/google/protobuf/stubs/common.h"
  cannot be read.
Call Stack (most recent call first):
  CMakeLists.txt:276 (find_package)

CMake Error at /usr/local/share/cmake-3.24/Modules/FindProtobuf.cmake:499 (math):
  math cannot parse the expression: " / 1000000": syntax error, unexpected
  exp_DIVIDE (2).
Call Stack (most recent call first):
  CMakeLists.txt:276 (find_package)

CMake Error at /usr/local/share/cmake-3.24/Modules/FindProtobuf.cmake:500 (math):
  math cannot parse the expression: " / 1000 % 1000": syntax error,
  unexpected exp_DIVIDE (2).
Call Stack (most recent call first):
  CMakeLists.txt:276 (find_package)

CMake Error at /usr/local/share/cmake-3.24/Modules/FindProtobuf.cmake:501 (math):
  math cannot parse the expression: " % 1000": syntax error, unexpected
  exp_MOD (2).
Call Stack (most recent call first):
  CMakeLists.txt:276 (find_package)

CMake Warning at /usr/local/share/cmake-3.24/Modules/FindProtobuf.cmake:524 (message):
  Protobuf compiler version 3.21.9 doesn't match library version
  ERROR.ERROR.ERROR
Call Stack (most recent call first):
  CMakeLists.txt:276 (find_package)
`

**Comment by briantylerec** [comment]
Hi @yonatan555 I'm facing the same problem, did you find a solution?

`CMake Warning at /usr/local/share/cmake-3.25/Modules/FindProtobuf.cmake:524 (message):
  Protobuf compiler version 3.21.9 doesn't match library version
  ERROR.ERROR.ERROR
Call Stack (most recent call first):
  CMakeLists.txt:26 (find_package)
`

**Comment by yonatan555** [comment]
Yeah, I will send set of commands to run when I will near the computer,  I am now from the phone

**Comment by yonatan555** [comment]
> Hi @yonatan555 I'm facing the same problem, did you find a solution?
> 
> `CMake Warning at /usr/local/share/cmake-3.25/Modules/FindProtobuf.cmake:524 (message): Protobuf compiler version 3.21.9 doesn't match library version ERROR.ERROR.ERROR Call Stack (most recent call first): CMakeLists.txt:26 (find_package) `

centos 7
GCC Compiler - 11.2.1
configure internet connection

// basic commands
yum -y update
sudo yum install wget -y && yum install perl-IPC-Cmd -y
yum install python3 -y 
sudo yum makecache 
sudo yum -y install pkgconfig
wget https://linuxpip.org/install-openssl-linux/
sudo yum install libcurl -y 
sudo yum install libcurl-devel -y

// git
sudo yum -y install https://packages.endpointdev.com/rhel/7/os/x86_64/endpoint-repo.x86_64.rpm 
sudo yum install git -y 

// compiler c++ 17
sudo yum group install "Development Tools" -y 
sudo yum install centos-release-scl -y 
sudo yum install devtoolset-11 -y
scl enable devtoolset-11 bash


configured open-ssl 1.1.1
// openssl
sudo yum install openssl-devel -y

// cmake
wget https://cmake.org/files/v3.24/cmake-3.24.0.tar.gz
scl enable devtoolset-11 bash
tar zxvf cmake-3.*
	cd cmake-3.*
	./bootstrap --prefix=/usr/local
	make -j$(nproc)
	make install

//gRPC
yum install -y build-essential autoconf libtool pkg-config
git clone -b v1.43.2 https://github.com/grpc/grpc
cd grpc
scl enable devtoolset-11 bash
git submodule update --init
mkdir -p cmake/build
pushd cmake/build
cmake -DgRPC_INSTALL=ON -DgRPC_BUILD_TESTS=OFF -DCMAKE_BUILD_TYPE=Release -DgRPC_ABSL_PROVIDER=module -DgRPC_CARES_PROVIDER=module -DgRPC_PROTOBUF_PROVIDER=module -DgRPC_RE2_PROVIDER=module -DgRPC_SSL_PROVIDER=module -DgRPC_ZLIB_PROVIDER=module ../..
make
make install
popd

// install OTLP
mkdir source && cd source && git clone --recursive https://github.com/open-telemetry/opentelemetry-cpp
git submodule update --init
scl enable devtoolset-11 bash
cd opentelemetry-cpp
mkdir build && cd build
cmake -DBUILD_TESTING=OFF -DWITH_EXAMPLES=ON -DWITH_OTLP=ON -DWITH_OTLP_GRPC=ON -DCMAKE_PREFIX_PATH=/root/grpc/ ..
cmake --build . --target all
cmake --install .
